Source/Core/DolphinQt/Updater.cpp

```diff
@@ -13,33 +13,58 @@
 
 #include "Common/Version.h"
 
+#include "DolphinQt/QtUtils/ModalMessageBox.h"
 #include "DolphinQt/QtUtils/RunOnObject.h"
 #include "DolphinQt/Settings.h"
 
 // Refer to docs/autoupdate_overview.md for a detailed overview of the autoupdate process
 
-Updater::Updater(QWidget* parent) : m_parent(parent)
+Updater::Updater(QWidget* parent, bool silently_fail)
+    : m_parent(parent), m_silently_fail(silently_fail)
 {
   connect(this, &QThread::finished, this, &QObject::deleteLater);
 }
 
 void Updater::run()
 {
-  AutoUpdateChecker::CheckForUpdate();
+  CheckForUpdate();
 }
 
-bool Updater::CheckForUpdate()
+void Updater::OnErrorOccurred(CheckError error)
 {
-  m_update_available = false;
-  AutoUpdateChecker::CheckForUpdate();
+  if (m_silently_fail)
+  {
+    return;
+  }
 
-  return m_update_available;
+  switch (error)
+  {
+  case CheckError::AlreadyUpToDate:
+    ModalMessageBox::information(
+        m_parent, tr("Update"),
+        tr("You are running the latest version available on this update track."));
+    break;
+  case CheckError::RequestFailed:
+    ModalMessageBox::critical(m_parent, tr("Update failed"),
+                              tr("Could not download update information. "
+                                 "Please check your Internet connection and try again."));
+    break;
+  case CheckError::InvalidJson:
+    ModalMessageBox::critical(m_parent, tr("Update failed"),
+                              tr("The server returned an invalid response. "
+                                 "Please check your Internet connection and try again."));
```

Comment:
I wonder if "Check your Internet connection" is a useful suggestion for "got an invalid response"...

I could only imagine a proxy blocking the request (where that suggestions _might_ remotely apply), or otherwise actual infrastructure issues (that the user probably can't influence anyways).

How about "Please try again in a few minutes" or something like that instead?

Reply:
I went with "Please try again later". That sounds more natural to me, but I'm fine with "in a few minutes" too.